### PR TITLE
Update dependency ignore list

### PR DESCRIPTION
**Pull request by mzfr**
Fix for issue #96

### CODE REVIEW

**Comment by enen92** [comment]
while you are at it maybe also add `inputstream.adaptive` and `script.module.pycryptodome`

**Comment by Rechi** [comment]
inputstream.adaptive is only v17 and higher and script.module.pycryptodome is v18 only, not sure if that should be handled somehow.

**Comment by Razzeee** [comment]
I would move the string we have right now to another method /file that accepts a branch and resolves it to a dependency list.

**Comment by mzfr** [comment]
Oh so we can have a different dependency list for all the different branch ?

**Comment by Razzeee** [comment]
Basically

**Comment by mzfr** [comment]
@Razzeee  I have made the changes please take a look

File: kodi_addon_checker/check_dependencies.py [hunk 1]
```diff
@@ -42,6 +47,21 @@ def check_addon_dependencies(report: Report, repo_addons: dict, parsed_xml):
...
 
+def _get_ignore_list(branch_name):
+
+    if branch_name == "leia":
+        common_ignore_deps.extend(["inputstream.adaptative", "inputstream.rtmp",
+                                   "script.module.cryptodome"])
+        return common_ignore_deps
+
+    elif branch_name == "krypton":
+        common_ignore_deps.extend(["inputstream.adaptative", "inputstream.rtmp"])
```
Comment (Razzeee) [inline kodi_addon_checker/check_dependencies.py]: Not a big fan of mutating lists in this way. I would have used `+` I guess. See fore more info: https://stackoverflow.com/a/252711/2859813

Comment (mzfr) [reply kodi_addon_checker/check_dependencies.py]: Just tested it and no it doesn't change with time.
Do you still want me to use the `+`

File: kodi_addon_checker/check_dependencies.py [hunk 2]
```diff
@@ -2,26 +2,31 @@
...
+                      'xbmc.python.pluginsource', 'xbmc.python.script', 'xbmc.python.weather', 'xbmc.python.lyrics',
+                      'xbmc.python.library', 'xbmc.python.module', 'xbmc.subtitle.module', 'kodi.context.item',
+                      'kodi.game.controller', 'xbmc.gui.skin', 'xbmc.webinterface', 'xbmc.addon.repository',
+                      'xbmc.pvrclient', 'kodi.gameclient', 'kodi.peripheral', 'xbmc.addon.video',
+                      'xbmc.addon.audio', 'xbmc.addon.image', 'xbmc.addon.executable', 'kodi.addon.game',
+                      'kodi.audioencoder', 'kodi.audiodecoder', 'xbmc.service', 'kodi.resource.images',
+                      'kodi.resource.language', 'kodi.resource.uisounds', 'kodi.resource.games',
+                      'kodi.resource.font', 'kodi.inputstream', 'kodi.vfs', 'kodi.imagedecoder', 'xbmc.json',
+                      'xbmc.gui', 'xbmc.json', 'xbmc.metadata', 'xbmc.python', 'script.module.pil',
+                      'inputstream.adaptive', 'script.module.pycryptodome']
```
Comment (Rechi) [inline kodi_addon_checker/check_dependencies.py]: shouldn't be added here

File: kodi_addon_checker/check_dependencies.py [hunk 3]
```diff
@@ -42,6 +47,21 @@ def check_addon_dependencies(report: Report, repo_addons: dict, parsed_xml):
                                   % (required_addon, required_version, available_version)))
 
 
+def _get_ignore_list(branch_name):
+
+    if branch_name == "leia":
+        common_ignore_deps.extend(["inputstream.adaptative", "inputstream.rtmp",
```
Comment (Rechi) [inline kodi_addon_checker/check_dependencies.py]: `inputstream.adaptive`

File: kodi_addon_checker/check_dependencies.py (hunk 1 above)
Comment (Rechi) [inline kodi_addon_checker/check_dependencies.py]: `inputstream.adaptive`

File: kodi_addon_checker/check_dependencies.py (hunk 2 above)
Comment (Razzeee) [reply kodi_addon_checker/check_dependencies.py]: https://github.com/xbmc/addon-check/commit/862de90c1ba3b7c1afff310af92802979cb32bb0

Comment (Rechi) [reply kodi_addon_checker/check_dependencies.py]: `script.module.pycryptodome` is also wrong here

File: kodi_addon_checker/check_dependencies.py [hunk 4]
```diff
@@ -42,6 +47,21 @@ def check_addon_dependencies(report: Report, repo_addons: dict, parsed_xml):
...
+                                   "script.module.cryptodome"])
```
Comment (Rechi) [inline kodi_addon_checker/check_dependencies.py]: `script.module.pycryptodome`

**Pull request merged**